WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -48,11 +55,17 @@ public void onCreate(Bundle savedInstanceState) {
         } else {
             mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
         }
+
+        if (mSite == null) {
+            mSite = mSiteStore.getSiteByLocalId(AppPrefs.getSelectedSite());
```

Comment:
I had meant to bring this up. The reason I added this is for the post upload completion notification, which launches the `PostsListActivity` to [display an alert](https://github.com/wordpress-mobile/WordPress-Android/blob/4c91f0911c149c804d59eac65164ae4255eb088b/WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java#L83-L101).

The usual ways of getting a reference to the site (above this line) fail in this case since the activity was launched with an [intent from the notification](https://github.com/wordpress-mobile/WordPress-Android/pull/4574/files#diff-95f8e47d54babc0d88147f3d2c9a89bbR115).

We could pass a reference to the model in the intent, but that would mean that if the user has switched sites since uploading and taps the notification, they'll be switched back to the previous site they were on. So, this seemed the most painless way to get the latest selected site.

Is there a better way of doing this?